tests/db/conftest.py

```
@@ -30,23 +30,22 @@ def setup_database(request, node_config):
     db_name = node_config['database']['name']
     conn = connect()
 
-    if conn.run(r.db_list().contains(db_name)):
-        conn.run(r.db_drop(db_name))
-
     try:
-        schema.init_database()
+        schema.init_database(conn)
     except DatabaseAlreadyExists:
         print('Database already exists.')
+        schema.drop_database(conn, db_name)
+        schema.init_database(conn)
 
     print('Finishing init database')
 
     def fin():
-        print('Deleting `{}` database'.format(db_name))
         conn = connect()
+        print('Deleting `{}` database'.format(db_name))
         try:
-            conn.run(r.db_drop(db_name))
-        except r.ReqlOpFailedError as e:
-            if e.message != 'Database `{}` does not exist.'.format(db_name):
+            schema.drop_database(conn, db_name)
+        except DatabaseDoesNotExist as e:
+            if str(e) != 'Database `{}` does not exist'.format(db_name):
                 raise
```

With the more granular exception, maybe we could just ignore this exception:

```python
except DatabaseDoesNotExist:
    pass
```